### DISCUSSION TITLE
404 error on HTTP API with /v1/config/paths/add/*name request

### DISCUSSION THREAD

**Discussion by brandousov** [bluenviron/mediamtx#688] [Questions]
## Which version are you using?

v0.17.6

## Which operating system are you using?

OS

- [x] Linux
- [ ] Windows
- [ ] macOS

Architecture

- [x] amd64
- [ ] arm64
- [ ] arm7
- [ ] arm6

## Describe the issue

Description

## Describe how to replicate the issue

1. start the server with defaults and "api: on" option.
2. making request from PHP to `http://127.0.0.1:9997/v1/config/paths/add/zzz` with payload `{"source":"rtsp://***:7554/p2-1", "sourceProtocol":"tcp"}`
3. Receive response 404: `array ( 0 => 'HTTP/1.0 404 Not Found', 'Content-Type' => 'text/plain', 'Server' => 'rtsp-simple-server', 'Date' => 'Fri, 22 Oct 2021 10:18:30 GMT', 'Content-Length' => '18', )`

```php
# v2
# REST API request to backend
# @help: https://aler9.github.io/rtsp-simple-server/#operation/configPathsAdd
# @param string $path URL for request to
# @param Array $data API query array data
function rtspApiRequest($path, $data = '')
{
	$content	=	f_isAry($data) ? http_build_query($data) : $data;
	$length		=	strlen($content);
	
	$result		=	get_headers(RTSP_API_URL . $path, true, stream_context_create([
		'http'	=>	[
			'method'	=>	'POST',
			'timeout'	=>	1,
			'header'	=>	"Content-Length: $length\r\nContent-Type: application/javascript\r\n",
			'content'	=>	$content,
			'user_agent' => 'icMediaApi',
			//'ignore_errors' => true
		]
	]));
	
	return $result;
}
```

**Comment by brandousov** [comment]
PS:
I think if we use String value for `encryption: "no"` parameter, then we must use strings for all other yes/no params. It warns after update to 0.17.6 that `boolean value can not be a string when json.unmarshal`..

**Comment by aler9** [comment]
Hello, if i run

```
curl -d '{}' http://127.0.0.1:9997/v1/config/paths/add/zzz
```

The server accepts the request without any problem:

```
2021/10/27 19:01:36 INF [0/0] [API] [conn 127.0.0.1:55240] POST /v1/config/paths/add/zzz
2021/10/27 19:01:36 INF [0/0] reloading configuration (API request)
2021/10/27 19:01:36 INF [0/0] [path zzz] created
```

Please make sure that the method (POST) is set correctly in your PHP library.

> I think if we use String value for encryption: "no" parameter, then we must use strings for all other yes/no params. It warns after update to 0.17.6 that boolean value can not be a string when json.unmarshal..

`encryption` was not meant to be a boolean param, since it can accept `no`, `optional`, `strict` (3 values) and is a string. The previous configuration parser accepted both booleans and strings and converted them into strings. The new parser is way better (it detects invalid keys, preventing a lot of users from opening issues) but it wants strings in case of string parameters, and bools in case of bool parameters.

[a comment by an automated account is not shown]
